### ISSUE TITLE
Cannot focus chat using keyboard in networking lobby

### ISSUE THREAD

**Issue by deveee** [supertuxkart/stk-code#3640]
It's especially visible when you are not server owner and there is no "start" button.

To fix chat with keyboard-only you can do for example:
```
diff --git a/src/guiengine/event_handler.cpp
b/src/guiengine/event_handler.cpp
index 9024853ee..72a8f2590 100644
--- a/src/guiengine/event_handler.cpp
+++ b/src/guiengine/event_handler.cpp
@@ -708,19 +708,8 @@ int EventHandler::findIDClosestWidget(const NavigationDirection nav, const int p
      // This allows to skip over disabled/invisible widgets in a grid
      if (!w_test->isVisible() || !w_test->isActivated())
      {
-        // Can skip over at most 3 consecutive disabled/invisible widget
-        if (recursion_counter <=2)
-        {
-            recursion_counter++;
-            return findIDClosestWidget(nav, playerID, w_test, /*ignore disabled*/ false, recursion_counter);
-        }
-        // If nothing has been found, do a search ignoring disabled/invisible widgets,
-        // restarting from the initial focused widget (otherwise, it could lead to weird results)
-        else if (recursion_counter == 3)
-        {
-            Widget* w_focus = GUIEngine::getFocusForPlayer(playerID);
-            return findIDClosestWidget(nav, playerID, w_focus, /*ignore disabled*/ true, recursion_counter);
-        }
+        Widget* w_focus = GUIEngine::getFocusForPlayer(playerID);
+        return findIDClosestWidget(nav, playerID, w_focus, /*ignore disabled*/ true, recursion_counter);
      }

      return closest_id;
```

Or even better:
```
diff --git a/src/guiengine/event_handler.cpp
b/src/guiengine/event_handler.cpp
index 9024853ee..d77ded29c 100644
--- a/src/guiengine/event_handler.cpp
+++ b/src/guiengine/event_handler.cpp
@@ -565,7 +565,7 @@ int EventHandler::findIDClosestWidget(const NavigationDirection nav, const int p
          // - it corresponds to an invisible or disabled widget
          // - the player is not allowed to select it
          if (w_test == NULL || !Widget::isFocusableId(i) || w == w_test ||
-            (!w_test->isVisible()   && ignore_disabled) ||
+            (!w_test->isVisible()) ||
              (!w_test->isActivated() && ignore_disabled) ||
              (playerID != PLAYER_ID_GAME_MASTER && !w_test->m_supports_multiplayer))
              continue;
```

But I don't know the reason why invisible widgets are not totally ignored. @Alayan-stk-2 Could you have a look?

**Comment by Alayan-stk-2** [comment]
When I developed this code, one of the screen I tested against was the options screen with checkboxes being made invisible if the connect to the internet option was unchecked.

You had a grid like this : 

[a1] [a2]
[d1] [a3]
[a4] [a5]

When on a1, we want down to go to a4. Without that code, it would jump to a3 (in most places, finding the widget closest horizontally is not the correct behavior ; finding the one closest vertically is better, so this is the default). The invisible/disabled widget is used as a stepping stone to find the next correct widget, a4 in that instance.
